packages/instantsearch.js/src/connectors/trending-items/connectTrendingItems.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  createDocumentationMessageGenerator,
+  checkRendering,
+  noop,
+} from '../../lib/utils';
+
+import type { Connector, TransformItems, Hit, BaseHit } from '../../types';
+import type {
+  PlainSearchParameters,
+  RecommendResultItem,
+} from 'algoliasearch-helper';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'trending-items',
+  connector: true,
+});
+
+export type TrendingItemsRenderState<THit extends BaseHit = BaseHit> = {
+  /**
+   * The matched recommendations from the Algolia API.
+   */
+  recommendations: Array<Hit<THit>>;
+};
+
+export type TrendingItemsConnectorParams<THit extends BaseHit = BaseHit> = {
+  /**
+   * The facet attribute to get recommendations for.
+   */
+  facetName?: string;
+  /**
+   * The facet value to get recommendations for.
+   */
+  facetValue?: string;
```

Comment:
This is not related to this PR (i think typing in the original recommend library has the same limitation) 

But for more correct typing, both `facetName` and `facetValue` have to be either set or not. We can't have one without the other or we will get an error: `Parameters 'facetName' and 'facetValue' must both be empty, or both be set`

Could be a future improvement though.

Reply:
I think it is safe to leave it to the API to throw an error in this case as we don't need (redundant) frontend validation, just some type checking would suffice.

Reply:
I found out how to type constraint it in https://github.com/algolia/instantsearch/pull/6169/commits/3c7ee20242f5804c321a6ee4641a34c5eb265cf2.

Reply:
nice !